src/pipeline/broad_phase/broad_phase.rs

```diff
@@ -24,6 +24,34 @@ impl ProxyHandle {
     }
 }
 
+/// Proximity handling for BroadPhase updates.
+pub trait BroadPhaseProximityHandler<T> {
+    /// A pre-filter that may cheaply discard objects before checking for bounding volume
+    /// interference.
+    fn allow(&mut self, &T, &T) -> bool;
+
+    /// Actually handle the interference.
+    /// The boolean specifies whether the objects start or stop interacting.
+    fn handle(&mut self, &T, &T, bool);
+}
+
+struct SimpleBroadPhaseProximityHandler<'a, T: 'a> {
```

Comment:
This can be removed as this is unnecessary if we don't keep the old `BroadPhase::update` method.